examples/wafv2-and-or-rules/main.tf

```diff
@@ -0,0 +1,106 @@
+provider "aws" {
+  region = "eu-west-1"
+}
+
+#####
+# Web Application Firewall configuration
+#####
+module "waf" {
+  source = "../.."
+
+  name_prefix = "test-waf-setup"
+  alb_arn     = aws_lb.test.arn
+
+  allow_default_action = true
+
+  scope = "REGIONAL"
+
+  create_alb_association = true
+
+  visibility_config = {
+    cloudwatch_metrics_enabled = false
+    metric_name                = "test-waf-setup-waf-main-metrics"
+    sampled_requests_enabled   = false
+  }
+
+  rules = [
+    {
+      name     = "AWSManagedRulesCommonRuleSet-rule-1"
+      priority = "1"
+
+      override_action = "none"
+
+      visibility_config = {
+        cloudwatch_metrics_enabled = false
+        metric_name                = "AWSManagedRulesCommonRuleSet-metric"
+        sampled_requests_enabled   = false
+      }
+
+      managed_rule_group_statement = {
+        name        = "AWSManagedRulesCommonRuleSet"
+        vendor_name = "AWS"
+        excluded_rule = [
+          "SizeRestrictions_QUERYSTRING",
+          "SizeRestrictions_BODY",
+          "GenericRFI_QUERYARGUMENTS"
+        ]
+      }
+    },
+    {
+      ### AND rule example
+      name     = "block-specific-uri-path-and-request-from-germany"
+      priority = 2
+      action   = "block"
+
+      and_statement = {
```

Comment:
as there can be multiple statement, you need to make it a list of objects

Reply:
Only one and / or statement per rule possible